### ISSUE TITLE
Incorrect Comparison in workorders when calculating is_produced blocking manufacturing

### ISSUE THREAD

**Issue by gdgellatly** [odoo/odoo#20919]
Impacted versions: 11.0

Steps to reproduce:
I found this on manufacturing a quantity of 43.8 using 3 digit UOM precision

From line 137 of mrp/mrp_workorder.py

```python
    @api.one
    @api.depends('production_id.product_qty', 'qty_produced')
    def _compute_is_produced(self):
        self.is_produced = self.qty_produced >= self.production_id.product_qty
```

must use float_compare or similar.  In this case we have 43.8 >= 43.80000000000000004

As a side note: self.qty_production is available on the workorder which is just a related field to self.production_id.product_qty

Current behavior:
A quantity of 0.00 is left remaining to be produced and it cannot be produced because it is 0.00

Expected behavior:
No remaining to be produced

Video/Screenshot link (optional):
![image](https://user-images.githubusercontent.com/5093264/32759693-a544a7c6-c94f-11e7-8977-26477542480b.png)

**Comment by gdgellatly** [comment]
Note there is another side effect to this, which is duplicating the production order leads to a div by zero error.  It is hard to tell if it is purely because of this bug or not, as a done manufacturing order does not reproduce it, and an in progress order with a quantity above 0 obviously doesn't do it either,

**Comment by Yenthe666** [comment]
@jco-odoo another issue in stock. Could you check please? (sorry for quite some pings).

**Comment by amoyaux** [comment]
Yep should use a float_compare. Gonna do a fix for it.

**Comment by gdgellatly** [comment]
@amoyaux did fix happen?

**Comment by amoyaux** [comment]
@gdgellatly Yep I did it. However I need a review and then I will merge it.

**Comment by amoyaux** [comment]
If you need the code already https://github.com/odoo/odoo/pull/21555

**Comment by amoyaux** [comment]
Should be fixed with commit fda755b885d00c2d8207f7662c4f95a51b30621e
Let me know if you still having issue with mrp
